test/mopidy.test.js

```diff
@@ -374,11 +401,20 @@ describe("._resetBackoffDelay", () => {
 
 describe(".close", () => {
   test("unregisters reconnection hooks", () => {
-    const spy = jest.spyOn(this.mopidy, "off");
+    const offSpy = jest.spyOn(this.mopidy, "off");
+    const reconnectingSpy = jest.fn();
+    this.mopidy.on("reconnecting", reconnectingSpy);
+    const reconnectionPendingSpy = jest.fn();
+    this.mopidy.on("reconnectionPending", reconnectionPendingSpy);
 
     this.mopidy.close();
 
-    expect(spy).toBeCalledWith("state:offline", this.mopidy._reconnect);
```

Comment:
This expectation was met with my original diff but the removeListener call was just silently failing, and reconnection would still be attempted on .close(). The new test failed on my original diff and passes on this one, because `this.mopidy._reconnect` is the method that gets registered with the EventEmitter.